eventstore/model.go

```diff
@@ -214,16 +224,25 @@ func (t *Txn) Save(updated ...interface{}) error {
 			return err
 		}
 
-		before := reflect.New(t.model.valueType.Elem()).Interface()
-		err = json.Unmarshal(beforeBytes, before)
-		if err != nil {
-			return err
+		var previous interface{}
+		if t.model.jsonMode {
+			// Instead of saving `before` as a Go Type of the before value of this instance
+			// lets just say that is exactly the json representation that is currently in
+			// the datastore.
+			previous = beforeBytes
+		} else {
+			before := reflect.New(t.model.valueType.Elem()).Interface()
+			err = json.Unmarshal(beforeBytes, before)
+			if err != nil {
+				return err
+			}
+			previous = before
```

Comment:
In JSONMode, `Action`  `Previous` and `Current` aren't Go Type representations of instances... they are just the json strings representations. So, here saved work for the store, no reflect magic and unmarshaling from datastore.